modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/PublisherIntegrationTestBase.cs

```diff
@@ -0,0 +1,213 @@
+﻿namespace Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Tests {
+    using Autofac;
+    using Microsoft.Azure.IIoT.Agent.Framework;
+    using Microsoft.Azure.IIoT.Diagnostics;
+    using Microsoft.Azure.IIoT.Hub;
+    using Microsoft.Azure.IIoT.Hub.Mock;
+    using Microsoft.Azure.IIoT.Hub.Models;
+    using Microsoft.Azure.IIoT.Module;
+    using Microsoft.Azure.IIoT.Module.Framework;
+    using Microsoft.Azure.IIoT.Module.Framework.Client;
+    using Microsoft.Azure.IIoT.Module.Framework.Hosting;
+    using Microsoft.Azure.IIoT.Module.Framework.Services;
+    using Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Agent;
+    using Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Controller;
+    using Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.Runtime;
+    using Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Engine;
+    using Microsoft.Azure.IIoT.OpcUa.Edge.Publisher.Models;
+    using Microsoft.Azure.IIoT.OpcUa.Protocol;
+    using Microsoft.Azure.IIoT.OpcUa.Protocol.Services;
+    using Microsoft.Azure.IIoT.Serializers;
+    using Microsoft.Azure.IIoT.Utils;
+    using Microsoft.Extensions.Configuration;
+    using Moq;
+    using Opc.Ua;
+    using Serilog;
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using static Microsoft.Azure.IIoT.Hub.Mock.IoTHubServices;
+
+    /// <summary>
+    /// Base class for integration testing, it connects to the server, runs publisher and injects mocked IoTHub services.
+    /// </summary>
+    public class PublisherIntegrationTestBase {
+        /// <summary>
+        /// Whether the module is running.
+        /// </summary>
+        private BlockingCollection<EventMessage> Events { get; set; } = new BlockingCollection<EventMessage>();
+
+        public PublisherIntegrationTestBase() {
+            // This is a fake but correctly formatted connection string.
+            var connectionString = $"HostName=dummy.azure-devices.net; DeviceId={Utils.GetHostName()};SharedAccessKeyName=iothubowner;SharedAccessKey=aXRpc25vdGFuYWNjZXNza2V5";
```

Comment:
even if this is only an example, please don't check something in that looks like credentials

Reply:
That's a fair point, I tried using plaintext in SharedAccessKey initially but unfortunately we have an existing code in mock class (IoTHubClientFactory) which verifies that the SharedAccessKey is a valid base64 string hence a valid string is used, in this case the above base64 string decodes to "itisnotanaccesskey". I wanted to keep the changes to minimal for existing code.

Reply:
ok